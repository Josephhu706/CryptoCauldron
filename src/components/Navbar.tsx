import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import logo from '../assets/images/logo.svg'
import Tabs from '@mui/material/Tabs';
import Tab from '@mui/material/Tab';
import {useState} from 'react'
import '../styles/NavbarStyles.scss'
import {useNavigate} from "react-router-dom";



export default function Navbar() {
    const [value, setValue] = useState(0);
    const navigate = useNavigate();

    const handleChange = (event: React.SyntheticEvent, newValue: number) => {
        setValue(newValue);
      };

    return (
        <Box sx={{ flexGrow: 1 }}>
        <AppBar position="static">
            <div className='navBar'>
                <Toolbar className='toolBar'>
                    <div className='logo' onClick={()=>navigate('/')}>
                        <img src={logo} alt="logo" />
                    </div>
                </Toolbar>
                <div className='navTabs'>
                    <Tabs onChange={handleChange} indicatorColor='secondary' textColor='inherit' value={value}>
                        <Tab label="Home" onClick={()=>navigate('/')}/>
                        <Tab label="Trending" onClick={()=>navigate('/trending')}/>
                    </Tabs>
                </div>
            </div>
        </AppBar>
        </Box>
    );
}