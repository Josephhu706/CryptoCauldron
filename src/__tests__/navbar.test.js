import {
  render,
  screen,
  cleanup,
  waitForElement,
  getByTestId,
  fireEvent,
} from "@testing-library/react";
import "@testing-library/jest-dom";
import Navbar from "../components/Navbar";
import renderer from "react-test-renderer";

// //makes sure each test is starting from the same starting point
afterEach(() => {
  cleanup();
});

const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

test("navbar should be rendered", () => {
  render(<Navbar />);
  const navbar = screen.getByTestId("navbar");
  expect(navbar).toBeInTheDocument();
});

test("clicking Home should navigate to homepage", () => {
  render(<Navbar />);
  const buttonEl = screen.getByTestId("home");
  fireEvent.click(buttonEl);
  expect(mockedUsedNavigate).toHaveBeenCalledWith(`/`);
});

test("clicking Trending should navigate to Trending", () => {
  render(<Navbar />);
  const buttonEl = screen.getByTestId("trending");
  fireEvent.click(buttonEl);
  expect(mockedUsedNavigate).toHaveBeenCalledWith(`/trending`);
});

test("clicking Logo should navigate to Home", () => {
  render(<Navbar />);
  const buttonEl = screen.getByTestId("logo");
  fireEvent.click(buttonEl);
  expect(mockedUsedNavigate).toHaveBeenCalledWith(`/`);
});

test("matches snapshot", () => {
  const tree = renderer.create(<Navbar />).toJSON();
  expect(tree).toMatchSnapshot();
});
