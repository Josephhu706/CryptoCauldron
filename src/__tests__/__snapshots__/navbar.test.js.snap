// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiBox-root css-i9gxme"
  data-testid="navbar"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="navBar"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular toolBar css-hyum1k-MuiToolbar-root"
      >
        <div
          className="logo"
          onClick={[Function]}
        >
          <img
            alt="logo"
            data-testid="logo"
            src="logo.svg"
          />
        </div>
      </div>
      <div
        className="navTabs"
      >
        <div
          className="MuiTabs-root css-1ujnqem-MuiTabs-root"
        >
          <div
            className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
            onScroll={[Function]}
            style={
              Object {
                "marginBottom": -0,
                "overflow": "hidden",
              }
            }
          >
            <div
              aria-orientation={null}
              className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
              onKeyDown={[Function]}
              role="tablist"
            >
              <button
                aria-selected={true}
                className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected css-cjctlb-MuiButtonBase-root-MuiTab-root"
                data-testid="home"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="tab"
                tabIndex={0}
                type="button"
              >
                Home
                <span
                  className="MuiTabs-indicator css-11yukd5-MuiTabs-indicator"
                  style={Object {}}
                />
              </button>
              <button
                aria-selected={false}
                className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit css-cjctlb-MuiButtonBase-root-MuiTab-root"
                data-testid="trending"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="tab"
                tabIndex={-1}
                type="button"
              >
                Trending
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;
